src/common/redis_module/ray_redis_module.cc

```diff
@@ -462,6 +460,46 @@ int TableAdd_RedisCommand(RedisModuleCtx *ctx,
       RedisModule_FreeString(ctx, publish_message);
       RedisModule_FreeString(ctx, publish_topic);
     }
+  } else if (pubsub_channel == TablePubsub_CLIENT) {
+    const char *buf = RedisModule_StringPtrLen(data, NULL);
+    auto client_data = flatbuffers::GetRoot<ClientTableData>(buf);
+
+    RedisModuleKey *clients_key = (RedisModuleKey *) RedisModule_OpenKey(
+        ctx, pubsub_channel_str, REDISMODULE_READ | REDISMODULE_WRITE);
+    // If this is a client addition, send all previous notifications, in order.
+    // NOTE(swang): This will go to all clients, so some clients will get
+    // duplicate notifications.
+    if (client_data->is_insertion() &&
+        RedisModule_KeyType(clients_key) != REDISMODULE_KEYTYPE_EMPTY) {
+      // NOTE(swang): Sets are not implemented yet, so we use ZSETs instead.
```

Comment:
That makes sense. If we want to remove the data structure from the server for the client table, we can go back to the design where a client scans the client table when it first subscribes. I just did not do that here since we don't have a "scan" API for the new GCS client.